File: clients/tools/src/models/Exercise.ts
import Tag from './Tag';

enum AnswerMethod {
    DANXUAN = "danxuan",
    DUOXUAN = "duoxuan",
    TIANKONG = "tiankong",
    JIEDA = "jieda",
    ZUOTU = "zuotu",
    JISUAN = "jisuan",
    ZONGHE = "zonghe",
    QITA = "qita"

}
interface Exercise {
    id?: string;
    content?: string; //习题内容
    answerMethod?: AnswerMethod;  //答题方式
    answer?: string; //答案
    explain?: string; //试题解析
    difficulty?: number; //难度系数
    tags?: Tag[];//试题标签
    outlineNodeId?: string;
}

export { Exercise, AnswerMethod }

File: aieduserver/src/main/kotlin/qx/aieduserver/models/Exercise.kt
package qx.aieduserver.models


/**
 * @author Parker
 * 试题数据对象

 * @param content 题目内容
 * @param answerMethod 试题的类型 danxuan,duoxuan,tiankong,jieda,zuotu,jisuan,zonghe
 * @param answer 试题答案内容
 * @param explain 试题解析
 * @param id 试题id
 *
 */
data class Exercise(
        var content: String?,
        var answerMethod: String? = null,
        var answer: String? = null,
        var explain: String? = null,
        var difficulty: Float? = null,
        var tags: MutableList<Tag>? = mutableListOf(),
        var id: String? = null)